### PR TITLE
Add build kit support

### PR DESCRIPTION
**Pull request by mr-smithers-excellent**
- Refactor `docker.js` to no longer extract inputs from `core`, instead passed in as parameters
- Refactor `docker-buld-push.test.js` by setting up default inputs & passing in overrides by test
- Create new `utils.js` to clean up helper functions
- Add buildKit support with new input `enableBuildKit`

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const dateFormat = require('dateformat');
+
+const timestamp = () => dateFormat(new Date(), 'yyyy-mm-dd.HHMMss');
+
+const parseArray = stringArray => {
```
Comment (scraggo) [inline src/utils.js:5]: nit: I'd prefer naming like, "stringArray" makes me think it's type `string[]`:

 `const getArgumentsFromCommaDelimitedString = (commaDelimitedString)` ?

**Review by scraggo — COMMENTED** [review]
looks good besides my nit!

File: src/utils.js (hunk 1 above)
Comment (mr-smithers-excellent) [reply src/utils.js:5]: I definitely was not in love with that naming convention - thanks for bringing it up!

**Pull request merged**